js/generate-images.mjs

```diff
@@ -102,8 +102,30 @@ export const deleteAll = () => {
   outputImages.splice(0, outputImages.length);
   renderOutput(outputImages);
   document.querySelector('#output-header').textContent =
-    'Output' +
-    (outputImages.length ? ' ( ' + outputImages.length + ' )' : '');
+    'Output' + (outputImages.length ? ' ( ' + outputImages.length + ' )' : '');
+};
+
+const arrayMove = (arr, oldIndex, newIndex) => {
+  if (newIndex >= arr.length) {
+    let k = newIndex - arr.length + 1;
+    while (k--) {
+      arr.push(undefined);
+    }
+  }
+  arr.splice(newIndex, 0, arr.splice(oldIndex, 1)[0]);
+  return arr; // for testing
+};
+
+export const moveLeft = (index) => {
+  if (index === 0) return outputImages;
```

Comment:
![image](https://user-images.githubusercontent.com/30949385/104817246-151b5580-5846-11eb-8627-424871e1a83b.png)
 I am getting this error when I click Right on the rightmost item. Any idea what could be causing this?

Reply:
Hey seems I did an off-by-one error `moveRight` function. Will correct it